src/shared/components/plots/MultipleCategoryHeatmap.tsx

```diff
@@ -141,33 +140,38 @@ function toHeatmapTracks(
         []
     );
 
-    const totalSamplesPerGroup = result.map(t =>
-        t.data
-            .map(d => d.profile_data)
-            .reduce((partialSum: number, d: number) => partialSum + d, 0)
-    ) as number[];
+    const groupsPercentages: number[][] = [];
+
+    groups.forEach(group => {
+        const groupSampleCount =
+            group.data
+                .map(d => d.profile_data)
+                .reduce((partialSum: number, d: number) => partialSum + d, 0) ||
+            0;
+        const groupPercentages: number[] = [];
+        groupsPercentages.push(groupPercentages);
+        group.data.forEach(d => {
+            const sampleCount = d.profile_data || 0;
+            const percentage = (sampleCount / groupSampleCount) * 100;
+            groupPercentages.push(percentage);
+            // Tooltip label:
+            d.sample = `${
+                d.sample
+            }; ${sampleCount} samples (${percentage.toFixed(1)}%)`;
+        });
+    });
 
-    const maxSamples = Math.max(
-        ...([] as any).concat(
-            ...result.map(t => t.data.map(d => d.profile_data))
-        )
+    // Calculate 'heat' of heatmap cell:
+    const percentageMax = Math.max(
+        ...groupsPercentages.reduce((group, result) => result.concat(group), [])
     );
+    groups.forEach((group, gi) => {
+        group.data.forEach((d, di) => {
+            d.profile_data = groupsPercentages[gi][di] / percentageMax;
```

Comment:
it seems odd that at line 154, "profile_data" is a sampleCount.   but here are making a percentage?

Reply:
the heatmap expects a value between 0 and 1, so sample count wouldn't work